timemachine/cpp/src/kernels/k_flat_bottom_bond.cuh

```diff
@@ -1,6 +1,65 @@
 #include "../fixed_point.hpp"
 #include "k_fixed_point.cuh"
 
+// branchless implementation of piecewise function
+template <typename RealType>
+RealType __device__ __forceinline__ compute_flat_bottom_energy(
+    RealType k, RealType r_lt_rmin, RealType r_gt_rmax, RealType r, RealType rmin, RealType rmax) {
```

Comment:
Don't immediately see need for intermediate variables `r_lt_rmin`, `r_gt_rmax` to be part of the signature, rather than computed from the other arguments -- can these be removed from the signature?

(Expanding in this way looks unsafe. For example, a user of this function can pass in `r, r_min, r_lt_rmin`, such that `r_lt_min != (r < rmin)` but no exception etc. would be surfaced.)

Reply:
https://github.com/proteneer/timemachine/pull/915/commits/3065e28567d3b47c0ac85f00be856e3847f4ad2e